### PR TITLE
Feature/token proxy

### PR DESCRIPTION
**Pull request by ThasianX**

### CODE REVIEW

**Comment by ThasianX** [comment]
> @Agunderman10 and I believe that we don't need proxy functions as well as base token. This added complexity should be removed in the PR. If we do need these we will add them with a future PR, since they are tangential.

good call

**Review by sagefarrenholz — APPROVED** [review]
nice job kev

File: programs/splitcoin-prism/src/context.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        payer = admin_authority,
+    )]
+    pub prism_asset: Account<'info, PrismAsset>,
+    /// Authority that has admin rights over the [PrismAsset].
+    pub admin_authority: Signer<'info>,
+    /// [Mint] of the [PrismAsset].
+    pub asset_mint: Account<'info, Mint>,
+    /// The [System] program.
+    pub system_program: Program<'info, System>,
+}
```
Comment (sagefarrenholz) [inline programs/splitcoin-prism/src/context.rs:25]: We need to rename this function. The instruction should be like `add_asset` not `initalize` initialize should be reserve for 1 time process of setting up the overall program state, seems to be convention on Solana programss

**Review by sagefarrenholz — CHANGES_REQUESTED** [review]

File: programs/splitcoin-prism/src/context.rs (hunk 1 above)
Comment (ThasianX) [reply programs/splitcoin-prism/src/context.rs]: changed the instruction handler name but forgot to change the struct name. good one

**Review by Agunderman10 — APPROVED** [review]

**Pull request merged**